src/Storage/importTransaction.js

```diff
@@ -1,23 +1,26 @@
 const { cloneDeep } = require('lodash');
 const { InvalidTransactionObject } = require('../errors');
 const { is, dashToDuffs } = require('../utils');
-
+const { SECURE_TRANSACTION_CONFIRMATIONS_NB, UNCONFIRMED_TRANSACTION_STATUS_CODE } = require('../CONSTANTS');
+const { FETCHED_UNCONFIRMED_TRANSACTION, FETCHED_CONFIRMED_TRANSACTION } = require('../EVENTS');
 /**
  * This method is used to import a transaction in Store.
  * @param transaction - A valid Transaction
  */
-const importTransaction = function (transaction) {
+const importTransaction = function importTransaction(transaction) {
   const self = this;
   if (!is.transactionObj(transaction)) throw new InvalidTransactionObject(transaction);
-  const transactionStore = this.store.transactions;
+  const { store } = this;
+
+  const transactionStore = store.transactions;
+  const currBlockheight = store.chains[this.network].blockheight;
```

Comment:
I think if we use this/self, it should be decomposed in the beginning of the function, so dependencies are more transparent.

Reply:
Resolved.